File: frontend/src/app/services/subtaskService/subtask.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { Subtask } from "src/app/models/subtask";

@Injectable({
  providedIn: "root"
})
export class SubtaskService {
  constructor(private https: HttpClient) {}
  private baseUrl = "/api/todos/subtask";

  saveSubtaskToATodo(subtask: Subtask): Observable<Subtask> {
    return this.https.post<Subtask>(this.baseUrl, subtask);
  }

  findAllSubtask(): Observable<Subtask[]> {
    return this.https.get<Subtask[]>(this.baseUrl);
  }
}

File: todo/src/main/java/com/todoapp/todo/service/SubtaskService.java
package com.todoapp.todo.service;

import java.util.List;

import com.todoapp.todo.entity.Subtask;

public interface SubtaskService {

	public Subtask saveSubTaskToATodo(Subtask subtask);
	
	public List<Subtask> findAllSubtask();
}
